server/src/main/java/io/crate/execution/dml/DoubleIndexer.java

```diff
@@ -63,14 +66,22 @@ public void indexValue(Number value,
                            Map<ColumnIdent, ColumnConstraint> toValidate) throws IOException {
         xcontentBuilder.value(value);
         double doubleValue = value.doubleValue();
-        if (ref.hasDocValues()) {
+        if (ref.hasDocValues() && ref.indexType() != IndexType.NONE) {
             addField.accept(new DoubleField(name, doubleValue));
         } else {
-            addField.accept(new DoublePoint(name, doubleValue));
-            addField.accept(new Field(
-                FieldNamesFieldMapper.NAME,
-                name,
-                FieldNamesFieldMapper.Defaults.FIELD_TYPE));
+            if (ref.indexType() != IndexType.NONE) {
+                addField.accept(new DoublePoint(name, doubleValue));
+            }
+            if (ref.hasDocValues()) {
```

Comment:
If i understand it correctly the first if L69 is both `docValues` and `index on`, then only `index on`, here we have only docvalues (`index off`) and finally none of the two.
It's more of a personal preference, but how about using a top level `else if` pattern?

Reply:
Yes right. But I also just figured out that the `<Number>Field` implementations do additionally support stored fields, the ctor's without declaring stored fields support are even deprecated. Such I added the `stored` logic also to them which keeps a top-level branching even more worth. See my last commit.

Reply:
I see, thx!